pkg/azure/resources.go

```diff
@@ -148,6 +172,40 @@ func ReadAndProvisionResource(ctx context.Context, resource *paragliderpb.Create
 	return handler.readAndProvisionResource(ctx, resource, subnet, resourceInfo, sdkHandler, additionalAddressSpaces)
 }
 
+func DoesVnetOverlapWithParaglider(ctx context.Context, handler *AzureSDKHandler, vnetName string, server *azurePluginServer) (bool, error) {
```

Comment:
@smcclure20  I'm thinking this function belongs to a different file. (we're passing in the azurePluginServer here). Thoughts? Or is it fine here?

Reply:
Hm, yeah this might be better somewhere else. Maybe just in utils?